src/Lib/Managers/NobManager.vala

```diff
@@ -0,0 +1,345 @@
+/*
+* Copyright (c) 2019 Alecaddd (http://alecaddd.com)
+*
+* This file is part of Akira.
+*
+* Akira is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+
+* Akira is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+
+* You should have received a copy of the GNU General Public License
+* along with Akira.  If not, see <https://www.gnu.org/licenses/>.
+*
+* Authored by: Giacomo Alberini <giacomoalbe@gmail.com>
+*/
+
+public class Akira.Lib.Managers.NobManager : Object {
+    /*
+        Grabber Pos:   8
+                     0 1 2
+                     7   3
+                     6 5 4
+
+        // -1 if no nub is grabbed
+    */
+    public enum Nob {
+        NONE=-1,
+        TOP_LEFT,
+        TOP_CENTER,
+        TOP_RIGHT,
+        RIGHT_CENTER,
+        BOTTOM_RIGHT,
+        BOTTOM_CENTER,
+        BOTTOM_LEFT,
+        LEFT_CENTER,
+        ROTATE
+    }
+
+    public weak Goo.Canvas canvas { get; construct; }
+
+    private Goo.CanvasItem root;
+    private Goo.CanvasItem select_effect;
+    private Goo.CanvasItemSimple[] nobs = new Goo.CanvasItemSimple[9];
+    private Goo.CanvasBounds select_bb;
+    private Nob selected_nob;
+    private double top;
+    private double left;
+    private double width;
+    private double height;
+    private double nob_size;
+    private double current_scale = 1.0;
+
+    public NobManager (Goo.Canvas canvas) {
+        Object (
+            canvas: canvas
+        );
+    }
+
+    construct {
+        root = canvas.get_root_item ();
+
+        event_bus.selected_items_changed.connect (on_add_select_effect);
+        event_bus.zoom.connect (on_zoom);
+    }
+
+    public void set_selected_by_name (Nob selected_nob) {
+        this.selected_nob = selected_nob;
+    }
+
+    public Nob get_selected_nob () {
+        return selected_nob;
+    }
```

Comment:
Don't use this approach but rather use the getter and setter methods when defining the variable in the class, like we were doing for the selected_item in the Canvas file.

Reply:
Actually, there's no need to have a public property baked by a private one, a simple public variable will work here!